### DISCUSSION TITLE
How to get whether block is transparent or not

### DISCUSSION THREAD

**Discussion by emortaldev** [Minestom/Minestom#755] [Help] [answered]
I want to check if the block type is a full cube

**Comment by emortaldev** [comment]
Using `block.registry().collisionShape()` and then checking relative start and end works to check if the block is a cube, but I'll keep the discussion open

**Comment by GoldenStack** [comment, marked as answer]
You can find out if a block is a full cube by getting its registry object (`Block#registry()`), getting its shape (`Registry.BlockEntry#collisionShape()`) and from that you can probably individually compare the properties of `Shape#relativeStart()` against `Shape#relativeEnd()` to see if it's exactly one block long in all directions.
